client/fakes/softlayer_client_fake.go

```diff
@@ -279,4 +297,6 @@ func (fslc *FakeSoftLayerClient) initSoftLayerServices() {
 	fslc.SoftLayerServices["SoftLayer_Billing_Item_Cancellation_Request"] = services.NewSoftLayer_Billing_Item_Cancellation_Request_Service(fslc)
 	fslc.SoftLayerServices["SoftLayer_Virtual_Guest_Block_Device_Template_Group"] = services.NewSoftLayer_Virtual_Guest_Block_Device_Template_Group_Service(fslc)
 	fslc.SoftLayerServices["SoftLayer_Hardware"] = services.NewSoftLayer_Hardware_Service(fslc)
+	fslc.SoftLayerServices["SoftLayer_Dns_Domain"] = services.NewSoftLayer_Dns_Domain_Service(fslc)
+	fslc.SoftLayerServices["SoftLayer_Dns_Domain_ResourceRecord"] = services.NewSoftLayer_Dns_Domain_Record_Service(fslc)
```

Comment:
The method name was derived from SL documentation (https://sldn.softlayer.com/reference/services/SoftLayer_Dns_Domain_ResourceRecord), but the service name was chosen just for reading convenience. However I agree with the statement on keeping it consistent and we have changed the name in https://github.com/TheWeatherCompany/softlayer-go/commit/4927e8b52a7c1d48e37694957f2cc63492de4b75 which will be pulled into this PR.